ncm-ssh/src/main/perl/ssh.pm

```diff
@@ -47,6 +51,15 @@ sub valid_sshd_file
     return 1;
 }
 
+# escape values for the ssh config format.
+# adds \ before ?{}.()[]
+sub ssh_escape
```

Comment:
this is potentially backwards incompatible. current profiles might have the escaped values as the actual value, this will make it worse. your pattern should not match any already escaped characters. adding negative lookbehind like 
```
$val =~ s{(?<!\\)([?{}.()\[\]])}{\\$1}g;
```
should avoid this, but this needs to be documented and is still potentially backwards incompatible.